### DISCUSSION TITLE
Ben Eater 8-Bit PC

### DISCUSSION THREAD

**Discussion by PaulKappmeyer** [hneemann/Digital#869] [Show and tell]
I’ve watched Ben Eaters 8-Bit computer series (yt: [here](https://www.youtube.com/watch?v=HyznrdDSSGM&list=PLowKtXNTBypGqImE405J2565dvjafglHU&ab_channel=BenEater)) and built his computer in Digital.

Since Digital already provides all the need 74xx ICs, it was very easy and fun to follow along. So, I wanted to share the result.
At first, I wanted to follow the built very strict meaning that the layout and connections should look like shown in the videos. So, I used every chip that was used in the video. This could be a bit inefficient, for example when just one inverter is used on a hex-inverter IC. 
Also, the wiring became a mess in the end (see first picture), which is way I decided to tweak the final build and share multiple versions. I have packed 4 Dig-Files and attached the ZIP-File at the end.

Afterwards you can find some explanations to the versions:

- ### 8-Bit PC 00:
This is the very complete version with every wire drawn out almost identical to the original.
<img src="https://user-images.githubusercontent.com/42696082/138289785-85d4dbed-ff50-4d52-9ee3-84cbb740cdd5.png" width=50%>

- ### 8-Bit PC 01:
This version uses splitters/mergers to make the bus connections and also tunnels on inputs and outputs to make the whole circuit more compact
<img src="https://user-images.githubusercontent.com/42696082/138289813-9158289a-1f0f-48ef-b6f2-3f2c5bf07ee9.png" width=50%>

- ### 8-Bit PC 02:
Even more tunnels are used to make the circuit more compact. And some basic logic ICs are replaced with the Digital gates to make their function clearer.
<img src="https://user-images.githubusercontent.com/42696082/138289846-158dbbfd-ea41-4ef8-b06a-30bea09808c6.png" width=50%>

**Program memory:**
The video series uses two 74189 ICs (64-Bit RAM, 16word 4-bit) as its program memory. DIP switches are used for manual programming. Anyway, this manual programming can get really annoying and exhausting. To be able to program the computer (a bit) easier, I slightly modified the 74189-chip, so that it now uses and EEPROM. This ROM can then be edited trough the ‘Content of ROMs’ menu.
<img src="https://user-images.githubusercontent.com/42696082/138290651-42903c26-15d7-4def-b374-0892399ed493.png" width=50%>
<img src="https://user-images.githubusercontent.com/42696082/138290671-7fad400e-0236-420b-9f0a-a073a0c548e7.png" width=50%>
<img src="https://user-images.githubusercontent.com/42696082/138290693-5990609f-86d5-4dea-b049-c2a2829293b1.png" width=50%>
<img src="https://user-images.githubusercontent.com/42696082/138290719-6cb40613-c633-4a2f-8515-7703326b8195.png">

- ### 8-Bit PC 03:
This version is the transmission of Ben Eaters PC to Digital without using the 74xx ICs. Instead, all functionality is provided by Digitals built-in components. This circuit is much more compact while containing the layout and functionality shown in the series.
**Program memory:**
A RAM component with the ‘Program memory’-option set is used as program memory. I’ve attached a hex file of a simple program, which counts to 255 and back down to 0. This file can be integrated to the preload option in the ‘Circuit specific settings’.
<img src="https://user-images.githubusercontent.com/42696082/138290790-ea157ae2-e46b-48ca-905a-ac43fcbcc5de.png" width=50%>

Here you can find all the Dig-Files and also the hex-File with the basic program:
[Ben_Eater_8_Bit_PC.zip](https://github.com/hneemann/Digital/files/7389628/Ben_Eater_8_Bit_PC.zip)

**Comment by ghost** [comment]
Nice!  I just started the video series and was about to start down the same path.

**Comment by Uprightfan** [comment]
Your schematics look great!  What is the software that made these files?  Can you show us a link to the app that was used?  

Thank you.

**Reply by PaulKappmeyer** [reply]
The software used is this projects itself. The schematics are screenshots out of Digital. You can find the files attached :)

**Reply by Uprightfan** [reply]
I found it!  I downloaded your Digital app, and it looks very good. Thank you.
